diff --git a/Cargo.toml b/Cargo.toml
index 6a46154..0e738f7 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -153,9 +153,11 @@ new_without_default = "allow"
 collapsible_match = "allow"
 
 [profile.release]
-debug = "line-tables-only"
-overflow-checks = true
-lto = "thin"
+opt-level = 3
+lto = "fat"
+codegen-units = 1
+panic = "abort"
+strip = true
 
 [profile.release.package.niri-config]
 # knuffel with chomsky generates a metric ton of debuginfo.
